sale_stock_picking_invoicing/models/stock_picking.py

```diff
@@ -0,0 +1,17 @@
+# Copyright (C) 2021-TODAY Akretion
+# @author Magno Costa <magno.costa@akretion.com.br>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def _get_partner_to_invoice(self):
+        partner_id = super()._get_partner_to_invoice()
+        if self.sale_id:
+            if partner_id != self.sale_id.partner_invoice_id.id:
+                partner_id = self.sale_id.partner_invoice_id.id
```

Comment:
in fact, you can simplify further to just:

```
        if self.sale_id:
            partner_id = self.sale_id.partner_invoice_id.id
```
if it's the same it will affect partner_id but it won't hurt performance and the code will be simpler

or eventually something like:

`partner_id = self.sale_id.partner_invoice_id.id if self.sale_id else super()._get_partner_to_invoice()`

Reply:
Done, the error in the tests is the problem we saw yesterday the Tests of module **account_invoice_pricelist** fail at a specific Date Time

Reply:
OK, the PR seems OK now. I will wait a few hours, trigger the tests again and approve once everything is green again. Thank you for the awesome work and sorry for the delay. Would be great if you could port it to 16.0 as well... (you can skip v15)

Reply:
I didn't saw your comment and just make a force push, I have tried to find where is the problem but not success, the clue is the problem related with Date Time, after UTC midnight the tests back to work

Reply:
@rvalyi I decide to make v15 migration https://github.com/OCA/account-invoicing/pull/1826 just to avoid the possibility of a problem for the cases of someone by some reason still using v15, after merge I'll port to v16